pandas/core/frame.py

```diff
@@ -646,9 +646,9 @@ def _constructor(self) -> Callable[..., DataFrame]:
         return DataFrame
 
     def _constructor_from_mgr(self, mgr, axes):
-        if self._constructor is DataFrame:
+        if type(self) is DataFrame:
```

Comment:
Does the test also pass if you keep the original `self._constructor is DataFrame` here?

Because I think that should fix the deprecation warning

Reply:
It does. I think I can also revert the changes to 

`DataFrame. _constructor_sliced_from_mgr ` and `Series. _constructor_expanddim_from_mgr` in this case

Reply:
Yes, indeed